modules/services/borgmatic.nix

```diff
@@ -0,0 +1,298 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.borgmatic;
+
+  mkNullableOption = args:
+    lib.mkOption (args // {
+      type = lib.types.nullOr args.type;
+      default = null;
+    });
+
+  mkRetentionOption = frequency:
+    mkNullableOption {
+      type = types.int;
+      description =
+        "Number of ${frequency} archives to keep. Use -1 for no limit.";
+      example = 3;
+    };
+
+  extraConfigOption = mkOption {
+    type = with types; attrsOf (oneOf [ str bool path int ]);
+    default = { };
+    description = "Extra settings.";
+  };
+
+  locationModule = types.submodule {
+    options = {
+      source_directories = mkOption {
+        type = types.listOf types.str;
+        description = "Directories to backup.";
+        example = literalExpression "[config.home.homeDirectory]";
+      };
+      repositories = mkOption {
+        type = types.listOf types.str;
+        description = "Paths to repositories.";
+        example =
+          literalExpression ''["ssh://myuser@myrepo.myserver.com/./repo"]'';
+      };
+      extraConfig = extraConfigOption;
+    };
+  };
+
+  storageModule = types.submodule {
+    options = {
+      encryption_passcommand = mkNullableOption {
+        type = types.str;
+        description = "Command writing the passphrase to standard output.";
+        example =
+          literalExpression ''"''${pkgs.password-store}/bin/pass borg-repo"'';
+      };
+      extraConfig = extraConfigOption;
+    };
+  };
+
+  retentionModule = types.submodule {
+    options = {
+      keep_within = mkNullableOption {
+        type = types.strMatching "[[:digit:]][Hdwmy]";
```

Comment:
How about "14d"? :-)

Reply:
Also, the format is different from `frequency` i the consistency check frequency?

Reply:
> How about "14d"?

you are perfectly right. Fixed and used "14d" as value for the unit test

> the format is different from frequency i the consistency check frequency?

this is expected. One configuration option comes from borg and the other one from borgmatic. Here is the reference documentation:
- https://torsion.org/borgmatic/docs/reference/configuration/
- https://borgbackup.readthedocs.io/en/stable/usage/prune.html